monai/networks/nets/dints.py

```diff
@@ -0,0 +1,556 @@
+        node_act_dict = {str(node_act_list[i]): i for i in range(len(node_act_list))}
+
+        self.num_depths = num_depths
+        self.filter_nums = filter_nums
+        self.cell_ops = cell_ops
+        self.code2in = code2in
+        self.code2ops = code2ops
+        self.code2out = code2out
+        self.node_act_list = node_act_list
+        self.node_act_dict = node_act_dict
+        self.trans_mtx = trans_mtx
+        self.tidx = tidx
+        self.child_list = np.array(child_list)
+        self.num_blocks = num_blocks
+        self.num_classes = num_classes
+        self.num_depths = num_depths
+        self.use_stem = use_stem
+
+        # define stem operations for every block
+        self.stem_down = nn.ModuleDict()
+        self.stem_up = nn.ModuleDict()
+        self.stem_finals = nn.Sequential(
+            nn.ReLU(),
+            nn.Conv3d(filter_nums[0], filter_nums[0], 3, stride=1, padding=1, bias=False),
+            nn.InstanceNorm3d(filter_nums[0]),
+            nn.Conv3d(filter_nums[0], num_classes, 1, stride=1, padding=0, bias=True),
+        )
+        for res_idx in range(num_depths):
+            if use_stem:
+                self.stem_down[str(res_idx)] = nn.Sequential(
+                    nn.Upsample(scale_factor=1 / (2 ** res_idx), mode="trilinear", align_corners=True),
+                    nn.Conv3d(in_channels, filter_nums[res_idx], 3, stride=1, padding=1, bias=False),
+                    nn.InstanceNorm3d(filter_nums[res_idx]),
+                    nn.ReLU(),
+                    nn.Conv3d(filter_nums[res_idx], filter_nums[res_idx + 1], 3, stride=2, padding=1, bias=False),
+                    nn.InstanceNorm3d(filter_nums[res_idx + 1]),
+                )
+                self.stem_up[str(res_idx)] = nn.Sequential(
+                    nn.ReLU(),
+                    nn.Conv3d(filter_nums[res_idx + 1], filter_nums[res_idx], 3, stride=1, padding=1, bias=False),
+                    nn.InstanceNorm3d(filter_nums[res_idx]),
+                    nn.Upsample(scale_factor=2, mode="trilinear", align_corners=True),
+                )
+
+            else:
+                self.stem_down[str(res_idx)] = nn.Sequential(
+                    nn.Upsample(scale_factor=1 / (2 ** res_idx), mode="trilinear", align_corners=True),
+                    nn.Conv3d(in_channels, filter_nums[res_idx], 3, stride=1, padding=1, bias=False),
+                    nn.InstanceNorm3d(filter_nums[res_idx]),
+                )
+                self.stem_up[str(res_idx)] = nn.Sequential(
+                    nn.ReLU(),
+                    nn.Conv3d(filter_nums[res_idx], filter_nums[res_idx], 3, stride=1, padding=1, bias=False),
+                    nn.InstanceNorm3d(filter_nums[res_idx]),
+                    nn.Conv3d(filter_nums[res_idx], num_classes, 1),
+                    nn.Upsample(scale_factor=2 ** res_idx, mode="trilinear", align_corners=True),
+                )
+
+        # define NAS search space
```

Comment:
`class DiNTS` becomes a complex python object (in terms of functionality). is it possible to decouple it into smaller pieces, for example, making this search space a standalone, small and compact pytorch nn.Module?